src/psyclone/tests/psyir/transformations/inline_trans_test.py

```diff
@@ -2438,3 +2471,48 @@ def test_apply_merges_symbol_table_with_routine(fortran_reader):
     inline_trans.apply(routine)
     # The i_1 symbol is the renamed i from the inlined call.
     assert psyir.walk(Routine)[0].symbol_table.get_symbols()['i_1'] is not None
+
+
+def test_apply_argument_clash(fortran_reader, fortran_writer):
+    '''
+    Check that the formal arguments to the inlined routine are not included
+    when checking for clashes (since they will be replaced by the actual
+    arguments to the call).
+    '''
+
+    code_clash = """
+  subroutine run_it()
+    integer :: Istr
+    real :: a(10)
+    istr = 2
+    a(Istr:10) = 1.0
+
+    call sub(Istr)
+  end subroutine run_it
+
+  subroutine sub(Istr)
+    integer :: Istr
+    real :: x
+    x = 2.0*x
+    call sub_sub(Istr)
+  end subroutine sub
+
+  subroutine sub_sub(Istr)
+    integer :: i
+    integer :: Istr
+    real :: b(10)
+
+    b(Istr:10) = 1.0
+  end subroutine sub_sub"""
+
+    psyir = fortran_reader.psyir_from_source(code_clash)
+    call = psyir.walk(Call)[1]
+    inline_trans = InlineTrans()
+    inline_trans.apply(call)
+    expected = '''\
+  x = 2.0 * x
+  b(istr:) = 1.0
+
+end subroutine sub
```

Comment:
Can you extend this to the whole subroutine, just to verify here that there isn't double declarations of any symbol.

Reply:
Oh yes, done. Also added a compilation check.